File: app/src/main/java/com/arn/ytscrobble/PrefFragment.java
package com.arn.ytscrobble;

import android.content.SharedPreferences;
import android.content.pm.PackageInfo;
import android.content.pm.PackageManager;
import android.os.Bundle;
import android.preference.Preference;
import android.preference.PreferenceFragment;
import android.preference.SwitchPreference;
import android.support.design.widget.AppBarLayout;
import android.support.design.widget.CollapsingToolbarLayout;

/**
 * Created by arn on 09/07/2017.
 */

public class PrefFragment extends PreferenceFragment implements SharedPreferences.OnSharedPreferenceChangeListener {

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setHasOptionsMenu(false);

        ((CollapsingToolbarLayout)getActivity().findViewById(R.id.toolbar_layout))
                .setTitle(getString(R.string.action_settings));
        AppBarLayout vg = (AppBarLayout) getActivity().findViewById(R.id.app_bar);
        vg.setExpanded(false, true);

        // Load the preferences from an XML resource
        addPreferencesFromResource(R.xml.preferences);
        Preference reauth = findPreference("reauth");
        reauth.setOnPreferenceClickListener(new Preference.OnPreferenceClickListener() {
            public boolean onPreferenceClick(Preference preference) {
                //open browser or intent here
                getPreferenceManager().getSharedPreferences()
                        .edit()
                        .remove("sesskey")
                        .apply();
                new Scrobbler(getActivity()).execute(Stuff.CHECKAUTH);
                return true;
            }
        });
        Preference about = findPreference("about");
        String pName = getActivity().getPackageName();
        try {
            PackageInfo pInfo = getActivity(). getPackageManager().getPackageInfo(pName, 0);
            String version = pInfo.versionName;
            about.setTitle(pName + " v" + version);
        } catch (PackageManager.NameNotFoundException e) {
            e.printStackTrace();
        }
    }


    @Override
    public void onSharedPreferenceChanged(SharedPreferences sp, String key) {
        if (key.equals("master")) {
            boolean val = sp.getBoolean("master", true);
            SwitchPreference mxm = (SwitchPreference) findPreference("scrobble_mxmFloatingLyrics"),
                    yt = (SwitchPreference) findPreference("scrobble_youtube");
            mxm.setChecked(val);
            yt.setChecked(val);
        }
    }

    @Override
    public void onResume() {
        super.onResume();
        getPreferenceManager().getSharedPreferences().registerOnSharedPreferenceChangeListener(this);

    }

    @Override
    public void onPause() {
        getPreferenceManager().getSharedPreferences().unregisterOnSharedPreferenceChangeListener(this);
        super.onPause();
    }

}

File: app/src/main/java/com/arn/ytscrobble/PrefFragment.kt
package com.arn.ytscrobble

import android.content.SharedPreferences
import android.content.pm.PackageManager
import android.os.Bundle
import android.preference.Preference
import android.preference.PreferenceFragment
import android.preference.SwitchPreference
import android.support.design.widget.AppBarLayout
import android.support.design.widget.CollapsingToolbarLayout

/**
 * Created by arn on 09/07/2017.
 */

class PrefFragment : PreferenceFragment(), SharedPreferences.OnSharedPreferenceChangeListener {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setHasOptionsMenu(false)

        (activity.findViewById(R.id.toolbar_layout) as CollapsingToolbarLayout).title = getString(R.string.action_settings)
        val vg = activity.findViewById(R.id.app_bar) as AppBarLayout
        vg.setExpanded(false, true)

        // Load the preferences from an XML resource
        addPreferencesFromResource(R.xml.preferences)
        val reauth = findPreference("reauth")
        reauth.title = reauth.title.toString() +": " + preferenceManager.sharedPreferences.getString("username", "nobody")
        reauth.onPreferenceClickListener = Preference.OnPreferenceClickListener {
            //open browser or intent here
            preferenceManager.sharedPreferences
                    .edit()
                    .remove("sesskey")
                    .apply()
            Scrobbler(activity).execute(Stuff.CHECKAUTH)
            true
        }
        val about = findPreference("about")
        val pName = activity.packageName
        try {
            val pInfo = activity.packageManager.getPackageInfo(pName, 0)
            val version = pInfo.versionName
            about.title = pName + " v" + version
        } catch (e: PackageManager.NameNotFoundException) {
            e.printStackTrace()
        }

    }


    override fun onSharedPreferenceChanged(sp: SharedPreferences, key: String) {
        if (key == "master") {
            val `val` = sp.getBoolean("master", true)
            val mxm = findPreference("scrobble_mxmFloatingLyrics") as SwitchPreference
            val yt = findPreference("scrobble_youtube") as SwitchPreference
            mxm.isChecked = `val`
            yt.isChecked = `val`
        }
    }

    override fun onResume() {
        super.onResume()
        preferenceManager.sharedPreferences.registerOnSharedPreferenceChangeListener(this)

    }

    override fun onPause() {
        preferenceManager.sharedPreferences.unregisterOnSharedPreferenceChangeListener(this)
        super.onPause()
    }

}
